db/statistics.go

```diff
@@ -932,219 +658,122 @@ func WriteValidatorDepositWithdrawals(day uint64, tx *sqlx.Tx) error {
 	}
 	lastSlot := utils.GetLastBalanceInfoSlotForDay(day)
 
-	start := time.Now()
-	logrus.Infof("Resetting Withdrawals + Deposits for day [%v]", day)
-
-	firstDayExtraCondition := ""
-	if day == 0 {
-		// genesis-deposits will be added to day -1.
-		firstDayExtraCondition = " OR day = -1"
-	}
+	logger := logger.WithFields(logrus.Fields{
+		"day":       day,
+		"firstSlot": firstSlot,
+		"lastSlot":  lastSlot,
+	})
 
-	resetQry := fmt.Sprintf(`
-		UPDATE validator_stats SET 
-			deposits = NULL, 
-			deposits_amount = NULL,
-			withdrawals = NULL, 
-			withdrawals_amount = NULL
-		WHERE day = $1%s;`, firstDayExtraCondition)
+	logger.Infof("gathering withdrawals + deposits")
 
-	_, err := tx.Exec(resetQry, day)
-	if err != nil {
-		return fmt.Errorf("error resetting deposit & withdrawal validator_stats for day [%v]: %w", day, err)
+	type resRowDeposits struct {
+		ValidatorIndex uint64 `db:"validatorindex"`
+		Deposits       uint64 `db:"deposits"`
+		DepositsAmount uint64 `db:"deposits_amount"`
 	}
-	logger.Infof("deposit & withdrawal reset completed, took %v", time.Since(start))
-
-	start = time.Now()
-	logrus.Infof("Update Withdrawals + Deposits for day [%v] slot %v -> %v", day, firstSlot, lastSlot)
-
-	logger.Infof("exporting deposits and deposits_amount statistics")
+	resDeposits := make([]*resRowDeposits, 0, 1024)
 	depositsQry := `
-		insert into validator_stats (validatorindex, day, deposits, deposits_amount) 
-		(
-			select validators.validatorindex, $3, count(*), sum(amount)
+			select validators.validatorindex, count(*) AS deposits, sum(amount) AS deposits_amount
 			from blocks_deposits
 			inner join validators on blocks_deposits.publickey = validators.pubkey
 			inner join blocks on blocks_deposits.block_root = blocks.blockroot
 			where blocks.slot >= $1 and blocks.slot <= $2 and blocks.status = '1' and blocks_deposits.valid_signature
-			group by validators.validatorindex
-		) 
-		on conflict (validatorindex, day) do
-			update set deposits = excluded.deposits, 
-			deposits_amount = excluded.deposits_amount;`
-	if day == 0 {
-		// genesis-deposits will be added to block 0 by the exporter which is technically not 100% correct
-		// since deposits will be added to the validator-balance only after the block which includes the deposits.
-		// to ease the calculation of validator-income (considering deposits) we set the day of genesis-deposits to -1.
```

Comment:
We have discussed this via DMs, here is a short recap:

Some kind of special handling will be required. And we have two related topics regarding "genesis deposits".

---

### Problem 1 - Missing deposits

Check for example https://sepolia.beaconcha.in/validator/0#deposits and https://prater.beaconcha.in/validator/0#deposits
=> No deposits are shown while https://beaconcha.in/validator/0#deposits has some.

This is most likely caused by
https://github.com/gobitfly/eth2-beaconchain-explorer/blob/a2b491e8836000b64c8c1ad7d98a0a54839d54bf/exporter/exporter.go#L736

---

### Problem 2 - Potential underflow

Once Genesis deposits are written for block 0, they will be written in the `day 0` row for each Genesis validator as the special handling this comment is linked too has been removed.

This would cause and underflow here though:
https://github.com/gobitfly/eth2-beaconchain-explorer/blob/f2150be871a460ae2de7cac36fbc1d0957968c21/db/statistics.go#L177
Remember, that `previousDayData.EndBalance = data.StartBalance` see https://github.com/gobitfly/eth2-beaconchain-explorer/blob/f2150be871a460ae2de7cac36fbc1d0957968c21/db/statistics.go#L157
With some example values taken from a validator on Mainnet:
`data.ClRewardsGWei = 32016114055 - 32000000000 + 0 - 32000000000`

---

Once it is decided how to tackle both issues, we should make sure that the handling is unified on all our supported networks. Maybe a db migration file will be required to move the deposit information from `validator_stats` with `day -1` to `day 0`.